var/spack/repos/builtin/packages/py-torchgeo/package.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyTorchgeo(PythonPackage):
+    """TorchGeo: datasets, transforms, and models for geospatial data.
+
+    TorchGeo is a PyTorch domain library, similar to torchvision, that provides
+    datasets, transforms, samplers, and pre-trained models specific to geospatial data.
+    """
+
+    homepage = "https://github.com/microsoft/torchgeo"
+    git = "https://github.com/microsoft/torchgeo.git"
+
+    maintainers = ['adamjstewart', 'calebrob6']
+
+    version('main', branch='main')
+
+    variant('datasets', default=False, description='Install optional dataset dependencies')
+    variant('docs', default=False, description='Install documentation dependencies')
+    variant('style', default=False, description='Install style checking tools')
+    variant('tests', default=False, description='Install testing tools')
+    variant('train', default=False, description='Install optional trainer dependencies')
+
+    # Required dependencies
+    depends_on('python@3.6:+bz2', type=('build', 'run'))
+    depends_on('py-setuptools@30.4:', type='build')
+    depends_on('py-einops', type=('build', 'run'))
+    depends_on('py-fiona@1.5:', type=('build', 'run'))
+    depends_on('py-matplotlib', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('pil@2.9:', type=('build', 'run'))
+    depends_on('py-pyproj@2.2:', type=('build', 'run'))
+    depends_on('py-rasterio@1.0.16:', type=('build', 'run'))
+    depends_on('py-rtree@0.5:', type=('build', 'run'))
+    depends_on('py-shapely@1.3:', type=('build', 'run'))
+    depends_on('py-torch@1.7:', type=('build', 'run'))
+    depends_on('py-torchvision@0.3:', type=('build', 'run'))
+
+    # Optional dependencies
+    with when('+datasets'):
```

Comment:
@alalazo I wasn't too keen on overloading `when` as a directive argument, decorator, and context manager at first, but this is pretty awesome!